src/fhirtypes/ElementDefinition.ts

```diff
@@ -524,11 +525,27 @@ export class ElementDefinition {
     // If element is a slice
     const slicedElement = this.slicedElement();
     if (slicedElement) {
-      // Check that slicedElement max >= new sum of mins
-      const sumOfMins = slicedElement.checkSumOfSliceMins(slicedElement.max, min - this.min);
-      // If new sum of mins > slicedElement min, increase slicedElement min
-      if (sumOfMins > slicedElement.min) {
-        slicedElement.min = sumOfMins;
+      const sliceSiblings = this.structDef.elements.filter(
+        el =>
+          this !== el &&
+          this.parentSliceName === el.parentSliceName &&
+          slicedElement === el.slicedElement()
+      );
+      const newParentMin = min + sliceSiblings.reduce((sum, el) => sum + el.min, 0);
+      // if this is a reslice, the parent element will also be a slice of the sliced element.
+      // if this is not a reslice, the parent element is the sliced element.
+      const parentElement = this.parentSliceName
+        ? this.structDef.elements.find(
+            el => el.sliceName === this.parentSliceName && slicedElement === el.slicedElement()
+          )
+        : slicedElement;
```

Comment:
Picking back up on the previous conversation about using `parentSliceName` to find the parent _vs_ using `/` in the `sliceName` to find the parent...  We've already established that `parentSliceName` will only be present for re-slices that were added by SUSHI in this profile.  So that means that if there are any re-slices in the parent profile, their elements will not have `parentSliceName`.  E.g., if base definition has slice `a` and re-slice `a/b`, and in FSH we re-slice `a/b` to make `a/b/c`, then we have:
* `a` with `parentSliceName === undefined`
* `a/b` with `parentSliceName === undefined` (since `a/b` is defined in the parent JSON)
* `a/b/c` with `parentSliceName === 'a/b'`

So... I think that means that when adding `a/b/c`, the cardinality constraints will properly be propagated from `a/b/c` to `a/b` (since `a/b/c` does have `parentSliceName`), but will not be properly propagated from `a/b` to `a` (since `a/b` does not have `parentSliceName`).

Similarly, if instead of reslicing `a/b` to create an `a/b/c`, you _added_ another reslice to `a`, making `a/b2`.  In this scenario, you would end up with:
* `a` with `parentSliceName === undefined`
* `a/b` with `parentSliceName === undefined` (since `a/b` is defined in the parent JSON)
* `a/b2` with `parentSliceName === 'a'`

Now we have a different problem in that in lines 528 - 533 above, the `a/b` re-slice will not be properly recognized as a sibling of `a/b2` since `a/b` does not have a `parentSliceName`.  This could also result in cardinality propagation errors since the sum of sibling slices would be incorrect.

The only way I see around this problem is to rely on using `/` to detect slice parents.  But if we need to do that anyway, then it seems like we would be overcomplicating things by attempting to use that approach _and_ `parentSliceName` at the same time.  It seems we should use `/` across the board.

Now, to mitigate things, we don't have to _only_ rely on `/`; we _can_ do some things to confirm that it is indeed a reslice.  If we see sliceName `x/y`, then there are a few things that will be true if it is a reslice:
* there will be an element w/ the same path and `sliceName === 'x'`
* that element will also have a `slicing` defined

If either of the above are not true, then `y` is _not_ a re-slice of `x`.

You may be able to keep much of this logic (and perhaps simplify some other logic) if you create a `getParentSlice` or `getParentSliceName` function that basically does what I described above. To support complex scenarios w/ multiple levels of re-slicing, I guess you would actually need to loop, walking the `/` backwards.  E.g., to find parent slice of `a/b/c/d`, you would first check:
* is `a/b/c` a valid slice w/ a slicing? If yes, it's the parent;
* if not, is `a/b` a valid slice w/ a slicing? If yes, it's the parent;
* if not is `a` a valid slice w/ a slicing? If yes, it's the parent;
* if not, this is not a re-slice, so it has not parent.